src/store/state.ts

```diff
@@ -43,6 +43,11 @@ export interface VQBState {
    */
   pipelines: { [name: string]: Pipeline };
 
+  /**
+   * the right dataset to be joined.
+   */
+  rightDataset: DataSet;
```

Comment:
I think having the `rightDataset` will matter only during the join step, when one must choose the column of the right dataset to join on.
So keepjng that info in the store may not be useful, as no other component will ever need it.

My recomendation, to keep the data as close as where we need it, would be to only keep the `rightDataset` in the local state (`data`) of the `JoinStepForm`.

What do you think of this idea?